service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -94,7 +108,15 @@ public void initializeSession(IoSession session, ConnectFuture future) {
             HttpConnectSession connectSession = (HttpConnectSession) session;
             connectSession.setVersion(acceptSession.getVersion());
             connectSession.setMethod(acceptSession.getMethod());
-            connectSession.setRequestURI(acceptSession.getRequestURI());
+            String acceptPathInfo = acceptSession.getPathInfo().toString();
+            if (acceptPathInfo.startsWith("/")) {
+                acceptPathInfo = acceptPathInfo.substring(1);
+            }
+            String connectPath = connectSession.getRequestURI().getPath();
+            if (!connectPath.endsWith("/")) {
```

Comment:
If the last segment from the connect path does not end in "/", then after .resolve() this last segment is removed.